adapter/utils/src/main/java/org/mobilitydata/gtfsvalidator/fileutils/CustomFileUtilsImpl.java

```diff
@@ -41,19 +42,43 @@ public static CustomFileUtilsImpl getInstance() {
     }
 
     /**
-     * Returns the size of a file given its path
+     * Returns the size of a file given its path in bytes
      * @param pathToFile  the path to the file as String
-     * @return the size of a file given its path
+     * @return the size of a file given its path in bytes
      */
     @Override
     public long sizeOf(final Path pathToFile) {
         return FileUtils.sizeOf(new File(pathToFile.toString()));
     }
 
     /**
-     * Returns the size of a directory given its path
+     * Returns the size of a file given its path in the specified unit. If no unit is specified, will return the result
+     * in the specified unit.
+     * @param pathToFile  the path to the file as String
+     * @param unit        the unit in which to express the result
+     * @return  the size of a file given its path in the specified unit. If no unit is specified, will return the result
+     * in the specified unit.
+     */
+    @Override
+    public long sizeOf(final Path pathToFile, final Unit unit) {
+        if (unit == null) {
+            throw new IllegalArgumentException("Unit must be provided");
```

Comment:
I would rather say "Unit must not be null if provided.", or similar, since there is a `sizeOf` method without the unit parameter.